independent-projects/bootstrap/runner/src/main/java/io/quarkus/bootstrap/runner/VirtualThreadSupport.java

```diff
@@ -0,0 +1,52 @@
+package io.quarkus.bootstrap.runner;
+
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+
+public class VirtualThreadSupport {
```

Comment:
I don't know if we already have some `quarkus-util` or `quarkus-common` module or anyway some utility module that is imported by all others, but if so I think that this class should be moved there. I've seen some other conditional behaviour based on the kind of the running thread spread across Quarkus and I believe that this will become even more frequent in a near future.

I think this is not urgent and maybe I could move this class somewhere else with a subsequent pull request, but in case you already know where I should move it please let me know.